### PR TITLE
DBZ-3575 Implement support for JSON_TABLE in MySQL parser

### PR DESCRIPTION
**Pull request by uurl**
[DBZ-3575 Implement support for JSON_TABLE in MySQL parser](https://issues.redhat.com/browse/DBZ-3575)

### CODE REVIEW

**Comment by uurl** [comment]
Ready @jpechane 😀

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 [hunk 1]
```diff
@@ -298,37 +297,40 @@ DAY_MICROSECOND:                     'DAY_MICROSECOND';
...
 JSON_STORAGE_FREE:                   'JSON_STORAGE_FREE';
 JSON_STORAGE_SIZE:                   'JSON_STORAGE_SIZE';
-JSON_ARRAYAGG:                       'JSON_ARRAYAGG';
-JSON_OBJECTAGG:                      'JSON_OBJECTAGG';
+JSON_TABLE:                          'JSON_TABLE';
+JSON_TYPE:                           'JSON_TYPE';
+JSON_UNQUOTE:                        'JSON_UNQUOTE';
+JSON_VALID:                          'JSON_VALID';
+JSON_VALUE:                          'JSON_VALUE';
+NESTED:                              'NESTED';
```
Comment (jpechane) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:331]: `NESTED` and `ORDINALITY` can be identifiers. Please check other tokens you added and add them to `keywordsCanBeId` rule if necessary.

**Review by jpechane — COMMENTED** [review]
@uurl Good start, thanks! Could you please take a look at my comments?

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 [hunk 2]
```diff
@@ -1115,8 +1116,40 @@ unionStatement
...
+               ('(' (querySpecificationNointo | queryExpressionNointo) ')' (AS? uid)?)
+              )
+    ;
+
+// JSON
+
+// https://dev.mysql.com/doc/refman/8.0/en/json-table-functions.html
+jsonTable
+    : JSON_TABLE '('
+        STRING_LITERAL ','
```
[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:1130]: @uurl Waa this comment really addressed?

[the remark this answers is not in the document]
Comment (uurl) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4]: Ready

**Comment by jpechane** [comment]
@uurl Applied, thanks a lot! Could you please also prepare a PR for upstream?

**Pull request merged**